### ISSUE TITLE
MagicOnObjectsPacket

### ISSUE THREAD

**Issue by IanWONeill** [chadalen/astraeus#31]
You missed one 👍

**Comment by chadalen** [comment]
@IanWONeill Thanks for letting me know.

Packet 35

**Comment by IanWONeill** [comment]
@IncomingPacket.IncomingPacketOpcode(35)
public class MagicOnObjectPacket implements Receivable  {
	
	@Override
	public void handlePacket(Player player, IncomingPacket packet) {
		ByteBufReader reader = packet.getReader();
		final int objectX = reader.readShort(ByteOrder.LITTLE);
	    final int spell = reader.readShort();
	    final int objectY = reader.readShort(false, ByteModification.ADDITION);
		final int objectId = reader.readShort(ByteOrder.LITTLE); 

		player.post(new MagicOnObjectEvent(objectId, objectX, objectY, spell));
	    
	    if (player.getRights().equal(PlayerRights.DEVELOPER) && player.attr().get(Player.DEBUG_KEY)) {
			player.queuePacket(new ServerMessagePacket("[MagicOnObject] used: " + spell + "/ on: " + objectId + ", x: " + objectX + ", y: " + objectY));
		}
	}

}

**Comment by chadalen** [comment]
@IanWONeill Thanks. The payload should look like this, I haven't tested though but looking at the client this is how you would read it.

```java

	@Override
	public void handlePacket(Player player, IncomingPacket packet) {

		ByteBufReader reader = packet.getReader();
		final int objectX = reader.readShort(ByteOrder.LITTLE);
	    final int spell = reader.readShort(false, ByteModification.ADDITION);
	    final int objectY = reader.readShort(false, ByteModification.ADDITION);
		final int objectId = reader.readShort(ByteOrder.LITTLE); 

		player.post(new MagicOnObjectEvent(objectId, objectX, objectY, spell));
	    
	    if (player.getRights().equal(PlayerRights.DEVELOPER) && player.attr().get(Player.DEBUG_KEY)) {
			player.queuePacket(new ServerMessagePacket("[MagicOnObject] used: " + spell + "/ on: " + objectId + ", x: " + objectX + ", y: " + objectY));
		}
		
	}
```

Can you send a pull request?

**Comment by IanWONeill** [comment]
The copy of Astreaus I have is renamed all over the place, I only started working on it yesterday. Also its a local source, im not connected through Github when working on it so I don't commit any changes to the branch I originaly made.

Theres quite alot of work to do with Astreaus, and finding a half decent client to work with it was half my problem. Though I intend to eventually develop something woth the time... Content isnt the issue, It's making a Framework to make the Content run better I feel. And what you have done here is almost what I'd consider what I'm looking for. (Obviously I've thrown my own flavour all over it since I started.)

Sorry though.
